x/mongo/driver/errors.go

```diff
@@ -377,7 +377,9 @@ func (e Error) NamespaceNotFound() bool {
 
 // ExtractErrorFromServerResponse extracts an error from a server response bsoncore.Document
 // if there is one. Also used in testing for SDAM.
-func ExtractErrorFromServerResponse(doc bsoncore.Document) error {
+//
+// Set isCSOT to true if "timeoutMS" is set on the Client.
+func ExtractErrorFromServerResponse(doc bsoncore.Document, isCSOT bool) error {
```

Comment:
Suggest not use a bool here and instead pass in the context and use csot.IsTimeoutContext(ctx) within decodeResult. This ensures that the source of CSOT-ness is derived from a context.

Reply:
Sounds good, will do.